src/components/ProfileForm/components/ProfileForm/ProfileFormNavigationButtons.tsx

```diff
@@ -8,33 +8,51 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 const ProfileFormNavigationButtons = ({
   onPrevious,
   onNext,
+  onFinish,
 }: {
   onPrevious?: () => void;
   onNext?: () => void;
+  onFinish?: () => void;
 }) => {
   return (
     <div className="flex justify-around m-4">
       {onPrevious !== undefined && (
+        <>
+          <Button
+            variant="full"
+            color="black"
+            size="medium"
+            onClick={onPrevious}
+            className="rounded-md"
+          >
+            <FontAwesomeIcon
+              icon={faAngleDoubleLeft}
+              className="mr-1 h-5 w-5"
+            />
+            Previous
+          </Button>
+        </>
+      )}
+      {onNext !== undefined && (
         <Button
+          type="submit"
           variant="full"
           color="black"
           size="medium"
-          onClick={onPrevious}
           className="rounded-md"
         >
-          <FontAwesomeIcon icon={faAngleDoubleLeft} className="mr-1 h-5 w-5" />
-          Previous
+          Next <FontAwesomeIcon icon={faAngleDoubleRight} className="h-5 w-5" />
         </Button>
       )}
-      {onNext !== undefined && (
+      {onFinish !== undefined && (
         <Button
-          type="submit"
           variant="full"
           color="black"
           size="medium"
+          onClick={onFinish}
           className="rounded-md"
         >
-          Next <FontAwesomeIcon icon={faAngleDoubleRight} className="h-5 w-5" />
+          Finish
         </Button>
```

Comment:
A couple of things in this file:

You pass the `onNext` callback but that's not used EXCEPT to decide if we show the next button or not.  This is weird, why pass a callback, instead it should be a boolean: `hasNext`.

For the `Finish` button, that's getting a new `onFinish` callback. BUT, is there a significant difference between Next and Finish? I don't think so. It's basically the same action, you can think of it as "continue", but we want different labels. So, I suggest you:

1. Drop both `onNext` and `onFinish`. Replace with one boolean: `hasNext`. If it's true, show the next button. Otherwise, show the Finish one. (we can infer that if it doesn't have a next step, it's the last one)
2. Don't duplicate code. Use the same button, but only change the content inside based on this prop. Also, the button should be `submit`.
3. Modify the code in the pare to work with this change